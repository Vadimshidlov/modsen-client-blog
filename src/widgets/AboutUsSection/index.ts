export { AboutUsSection } from '@/widgets/AboutUsSection/ui/AboutUsSection';
