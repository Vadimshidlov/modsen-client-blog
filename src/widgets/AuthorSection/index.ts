export { AuthorSection } from '@/widgets/AuthorSection/ui/AuthorSection';
