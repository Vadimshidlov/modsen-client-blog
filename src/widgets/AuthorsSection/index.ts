export { AuthorsSection } from '@/widgets/AuthorsSection/ui/AuthorsSection';
