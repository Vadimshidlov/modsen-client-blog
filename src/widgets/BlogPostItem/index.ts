export { BlogPostItem } from '@/widgets/BlogPostItem/BlogPostItem';
