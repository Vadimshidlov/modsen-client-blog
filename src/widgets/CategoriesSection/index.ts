export { CategoriesSection } from '@/widgets/CategoriesSection/ui/CategoriesSection';
