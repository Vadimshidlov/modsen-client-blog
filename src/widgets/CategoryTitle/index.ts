export { CategoryTitle } from '@/widgets/CategoryTitle/ui/CategoryTitle';
