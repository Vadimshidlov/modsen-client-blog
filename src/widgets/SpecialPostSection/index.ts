export { SpecialPostSection } from '@/widgets/SpecialPostSection/ui/SpecialPostSection';
