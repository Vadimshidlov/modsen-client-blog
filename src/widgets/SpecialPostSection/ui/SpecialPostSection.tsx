import React from 'react';
import Image from 'next/image';
import { SpecialPost } from '@/widgets/SpecialPostSection/ui/SpecialPost';
import specialPostImageSrc from '@/widgets/SpecialPostSection/assets/close-up-photography-of-man-wearing-sunglasses-1212984.jpg';
import container from '@/shared/styles/container.module.scss';
import styles from '@/widgets/SpecialPostSection/ui/SpecialPostSection.module.scss';

export function SpecialPostSection() {
  return (
    <section className={`${styles.specialPostSection} ${container.container}`}>
      <Image
        src={specialPostImageSrc}
        className={styles.postImage}
        placeholder="blur"
        alt="special-post-image"
      />
      <SpecialPost />
    </section>
  );
}
