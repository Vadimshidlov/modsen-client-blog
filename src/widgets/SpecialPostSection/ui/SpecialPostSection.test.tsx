import { SpecialPostSection } from '@/widgets/SpecialPostSection';
import React from 'react';
import { render } from '@testing-library/react';

jest.mock('next-intl', () => ({
  useTranslations: jest.fn().mockReturnValue((key: string) => `Mocked ${key}`),
}));

describe('SpecialPostSection tests', () => {
  test('should matches snapshot', () => {
    const { container } = render(<SpecialPostSection />);
    expect(container).toMatchSnapshot();
  });
});
