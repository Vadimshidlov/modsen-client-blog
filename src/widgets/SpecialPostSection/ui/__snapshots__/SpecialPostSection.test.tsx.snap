// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpecialPostSection tests should matches snapshot 1`] = `
<div>
  <section
    class="specialPostSection container"
  >
    <img
      alt="special-post-image"
      class="postImage"
      data-nimg="1"
      decoding="async"
      height="40"
      loading="lazy"
      src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
      srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
      style="color: transparent; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
      width="40"
    />
    <article
      class="contentBlock"
    >
      <h3
        class="postSubTitle variable"
      >
        Mocked title
      </h3>
      <h2
        class="postTitle"
      >
        Mocked postTitle
      </h2>
      <span
        class="postText"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip.
      </span>
      <button
        class="button"
        type="button"
      >
        <a
          href="/about"
        >
          Mocked buttonText
           &gt;
        </a>
      </button>
    </article>
  </section>
</div>
`;
